@ecomplus/storefront-components/src/js/BuyTogether.js

```diff
@@ -186,17 +243,24 @@ export default {
           if (!this.productIds.length) {
             if (this.relatedProducts.length) {
               this.setProductQnts(this.relatedProducts)
+              this.hasLoadedItems = true
+              this.ecomSearch.setProductIds(this.productIds)
+              this.fetchItems()
             } else {
               graphs({ url: `/products/${this.baseProduct._id}/related.json` }).then(({ data }) => {
                 this.setProductQnts(recommendedIds(data))
                 this.$nextTick(() => {
                   if (!this.productIds.length) {
                     this.hasLoadedItems = true
                   }
+                  this.ecomSearch.setProductIds(this.productIds)
+                  this.fetchItems()
                 })
               })
             }
           }
+          this.ecomSearch.setProductIds(this.productIds)
+          this.fetchItems()
```

Comment:
Tá fazendo requests (`this.fetchItems()`) duplicadas

Reply:
Faltou um else aí no caso..

Reply:
E notou que toda vez você repete:

```js
this.ecomSearch.setProductIds(this.productIds)
this.fetchItems()
```

?

Sua função `fetchItems` deveria conter o `this.ecomSearch.setProductIds(this.productIds)`, aí você não precisaria repetir tanto. Pela lógica nesse componente aqui o fetch é sempre pelo filtro dos `this.productIds` ...